### PR TITLE
Feature sort deliveries

### LINKED ISSUE #88: Sort Delivery Command Feature
Create sort delivery command which allows users to sort by ascending or descending order and field to sort by
Create relevant test cases to test the sort delivery command and comparators

### PR DESCRIPTION
**Pull request by ngjianyi**
Closes #88

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/SortDeliveryCommand.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * Sorts and lists all deliveries in address book by the field.
+ */
+public class SortDeliveryCommand extends SortCommand {
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " " + PREFIX_DELIVERY
+            + ": Sorts all deliveries by cost, date time or status and "
+            + "displays them as a list with index numbers.\n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_DELIVERY + " "
+            + PREFIX_SORT_ORDER + "SORT ORDER ('a' for ascending and 'd' for descending) "
+            + PREFIX_SORT_BY + "SORT BY ('s' for status, 'd' for datetime, 'c' for cost)";
```
Comment (CJianzhi) [inline src/main/java/seedu/address/logic/commands/SortDeliveryCommand.java:23]: Should there be a concrete example given in MESSAGE_USAGE?

File: src/main/java/seedu/address/logic/parser/SortDeliveryCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+public class SortDeliveryCommandParser implements Parser<SortDeliveryCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SortDeliveryCommand
+     * and returns a SortDeliveryCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public SortDeliveryCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_SORT_ORDER, PREFIX_SORT_BY);
+        //  if (!arePrefixesPresent(argMultimap, PREFIX_SORT_BY) || !argMultimap.getPreamble().isEmpty()) {
```
Comment (CJianzhi) [inline src/main/java/seedu/address/logic/parser/SortDeliveryCommandParser.java:26]: Should this commented out code be removed?

File: src/main/java/seedu/address/logic/parser/SortOrder.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    /**
+     * Sort order must be 'a' or 'd'.
+     */
+    public static final String VALIDATION_REGEX = "^[ad]$";
+    private final boolean isSortByAscending;
+
+    /**
+     * Creates a SortOrder object with the corresponding sorting order.
+     * @param sortOrder Represents ascending with 'a' and descending with 'd'.
```
Comment (CJianzhi) [inline src/main/java/seedu/address/logic/parser/SortOrder.java:24]: Should there be a line separating the description from params?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -151,9 +160,23 @@ public ObservableList<Delivery> getFilteredDeliveryList() {
     @Override
     public void updateFilteredDeliveryList(Predicate<Delivery> predicate) {
         requireNonNull(predicate);
+        isViewingFilteredList = true;
```
Comment (CJianzhi) [inline src/main/java/seedu/address/model/ModelManager.java:163]: The use of a boolean field to determine which list to display is brillaintly designed!

File: src/main/java/seedu/address/model/delivery/DeliverySortCostComparator.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof DeliverySortCostComparator)) {
+            return false;
+        }
+
+        return super.equals(other);
```
Comment (CJianzhi) [inline src/main/java/seedu/address/model/delivery/DeliverySortCostComparator.java:35]: Should this be super.equals(other.super) instead?

**Review by CJianzhi — APPROVED** [review]
LGTM! Well designed methods to sort deliveries by each field using Comparators. However, some JavaDocs comments are not in the correct format.

File: src/main/java/seedu/address/logic/parser/SortOrder.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+     */
+    public SortOrder(String sortOrder) {
+        requireNonNull(sortOrder);
+        checkArgument(isValidSortOrder(sortOrder), MESSAGE_CONSTRAINTS);
+        this.isSortByAscending = sortOrder.equals("a");
```
Comment (LinkesV) [inline src/main/java/seedu/address/logic/parser/SortOrder.java:29]: I think the sort order should be changed into a switch case in order to test properly for correct input and so we can practise defensive programming

**Review by LinkesV — APPROVED** [review]

**Review by LinkesV — APPROVED** [review]
LGTM!

**Pull request merged**